src/components/molecules/UserSearchBar/UserSearchBar.tsx

```diff
@@ -60,7 +54,9 @@ const UserSearchBar: FC<UserSearchBarProps> = ({ onSelect }) => {
             <div
               className="row result-user"
               key={user.id}
-              onClick={() => onSelect(user)}
+              onClick={() => {
+                setUserProfile(user);
+              }}
```

Comment:
Where does `onSelect` come from? And why are we not using it here anymore? Should we be removing other code related to `onSelect`? This feels like it may not be correct?

Also, don't need to wrap in the extra `{` `}`'s

Reply:
Same as OnlineStats it is not used anywhere. Again I am not sure.

Reply:
From a quick search, it seems like `UserSearchBar` and related aren't actually used by anything at the moment, so I assume they also relate to legacy code.

As a general rule though, since there is still an `onSelect` property being passed to the component, I would consider removing it (particularly when you don't know what it does/why) to be an unrelated change; so in this instance, I would probably expect it to still be called within this `onClick` handler. If we had better information (eg. this component was actually used somewhere) then we could make a better decision as to whether we should remove that prop entirely or not. But since we can't, I would say add the `onSelect(user)` call back here, as well as the `setUserProfile(user)` call.